### PR TITLE
Fixes error handling and slack error response messaging

### PR DESCRIPTION
**Pull request by keeganbrown**
# Why?
- Two different errors need to be handled -- a slack message that doesn't have a proper repo URL, and a slack message that contains a URL that can't be found. #26 

# What changed?
- Added error handling
- Added test for error handling

### CODE REVIEW

File: lib/crbot.js [hunk 1]
```diff
@@ -30,19 +30,28 @@ class CrBot {
...
+  onIncomingCodeReview(req, res, next) {
+    let githubPRParams = this.getGitHubPRParams(req.body.text);
+    let newCR;
+    if (!githubPRParams) {
+      return this.onSlackMessageIsBad(res, req);
     }
 
     return this.getPR(githubPRParams)
       .then(githubResponse => {
+        if ( !githubResponse[0] ) {
```
Comment (zachary-kuhn) [inline lib/crbot.js]: Returning false here will have it call `this.slack.createCRMessage(false);` down below, right?

Comment (keeganbrown) [inline lib/crbot.js]: Wow. Apparently I _thought_ about writing code, but didn't actually write it, and then submitted the PR. Please hold.

Comment (keeganbrown) [inline lib/crbot.js]: @zachary-kuhn Ok -- now this makes more sense. Please resume your CR :)

File: spec/crbot_spec.js [hunk 2]
```diff
@@ -100,12 +144,12 @@ describe(`${__filename.slice(__dirname.length + 1)}: CRBot Server`, () => {
...
       }).then((codeReview) => {
-        console.log(codeReview._created);
+        //console.log(codeReview._created);
         codeReview.emitter.on('killme', () => {
           done();
         }).catch((err) => {
-          console.log(err);
-          console.trace();
+          //console.log(err);
+          //console.trace();
```
Comment (greg5green) [inline spec/crbot_spec.js]: Can you just outright delete these commented lines

**Comment by greg5green** [comment]
Fine work Keegan. Fine work. :wave:

**Comment by keeganbrown** [comment]
Thank you kind sir.

**Pull request merged**